bindings/java/src/convert.rs

```diff
@@ -69,29 +69,33 @@ pub(crate) fn string_to_jstring<'a>(
     )
 }
 
-pub(crate) fn get_optional_string_from_object(
+pub(crate) fn read_bool_field(
     env: &mut JNIEnv<'_>,
     obj: &JObject,
-    method: &str,
+    field: &str,
+) -> crate::Result<bool> {
+    Ok(env.get_field(obj, field, "Z")?.z()?)
+}
+
+pub(crate) fn read_string_field(
```

Comment:
Some perf ideas:

* Filter out null options object (currently we always pass an option object, but can be null to indicate all default anyway)
* Cache field id and try to use `get_field_unchecked`.